adafruit_tlc59711/adafruit_tlc59711_multi.py

```diff
@@ -0,0 +1,1120 @@
+
+    def _debug_print_tlc59711_header(self):
+        # print(
+        #     "self.CHIP_BUFFER_HEADER_BYTE_COUNT",
+        #     self.CHIP_BUFFER_HEADER_BYTE_COUNT)
+        print("header: '", end="")
+        for index in range(self.CHIP_BUFFER_HEADER_BYTE_COUNT):
+            print("{:08b} ".format(self._buffer[index]), end="")
+        print("' ", end="")
+
+    def _debug_print_tlc59711_ch(self):
+        print("ch: [", end="")
+        for index in range(
+                self.CHIP_BUFFER_HEADER_BYTE_COUNT, len(self._buffer)
+        ):
+            print("x{:02X}, ".format(self._buffer[index]), end="")
+        print("]", end="")
+
+    ##########################################
+
+    def _get_32bit_value_from_buffer(self, buffer_start):
+        return (
+            (self._buffer[buffer_start + 0] << 24) |
+            (self._buffer[buffer_start + 1] << 16) |
+            (self._buffer[buffer_start + 2] << 8) |
+            self._buffer[buffer_start + 3]
+        )
+
+    def _set_32bit_value_in_buffer(self, buffer_start, value):
+        if not 0 <= value <= 0xFFFFFFFF:
+            raise ValueError(
+                "value {} not in range: 0..0xFFFFFFFF"
+                "".format(value)
+            )
+        # print("buffer_start", buffer_start, "value", value)
+        # self._debug_print_buffer()
+        self._buffer[buffer_start + 0] = (value >> 24) & 0xFF
+        self._buffer[buffer_start + 1] = (value >> 16) & 0xFF
+        self._buffer[buffer_start + 2] = (value >> 8) & 0xFF
+        self._buffer[buffer_start + 3] = value & 0xFF
+
+    def _get_16bit_value_from_buffer(self, buffer_start):
+        return (
+            (self._buffer[buffer_start + 0] << 8) |
+            self._buffer[buffer_start + 1]
+        )
+
+    def _set_16bit_value_in_buffer(self, buffer_start, value):
+        # if not 0 <= value <= 65535:
+        #     raise ValueError(
+        #         "value {} not in range: 0..65535"
+        #         "".format(value)
+        #     )
+        # print("buffer_start", buffer_start, "value", value)
+        # self._debug_print_buffer()
+        self._buffer[buffer_start + 0] = (value >> 8) & 0xFF
+        self._buffer[buffer_start + 1] = value & 0xFF
+
+    @staticmethod
+    def _convert_01_float_to_16bit_integer(value):
+        """Convert 0..1 Float Value to 16bit (0..65535) Range."""
+        # check if value is in range
+        if not 0.0 <= value[0] <= 1.0:
+            raise ValueError(
+                "value[0] {} not in range: 0..1"
+                "".format(value[0])
+            )
+        # convert to 16bit value
+        return int(value * 65535)
+
+    @classmethod
+    def _convert_if_float(cls, value):
+        """Convert if value is Float."""
+        if isinstance(value, float):
+            value = cls._convert_01_float_to_16bit_integer(value)
+        return value
+
+    @staticmethod
+    def _check_and_convert(value):
+        # check if we have float values
```

Comment:
Could rewrite this as a loop.

Reply:
tried this as loop. it is very slow. so i re-changed it to discreet version.
(if the lib is to slow to use with more than 1 chip there is no point in making a multi chip version ;-) )